app/recurring_job.go

```diff
@@ -514,12 +469,12 @@ func (job *Job) listSnapshotNamesForCleanup(snapshots []longhornclient.Snapshot,
 	// Only consider deleting the snapshots that were created by our current job
 	snapshots = filterSnapshotsWithLabel(snapshots, types.RecurringJobLabel, jobLabel)
 
-	if job.jobType == jobTypeSnapshot {
-		return filterExpiredItems(snapshotsToNameWithTimestamps(snapshots), job.retain)
+	if job.task == longhorn.RecurringJobTypeSnapshot {
+		return filterExpiredItems(snapshotsToNameWithTimestamps(snapshots, false), job.retain)
```

Comment:
Why not exclude the volume head here?

Reply:
The original intention is to keep the current logic to minimize the impact. 
Having a second look at this and seems cleanup will only remove the older snapshot, so the returned list will never be the volume head.

I'm not sure about the original purpose of why the volume head isn't excluded. Likely that we did not need the extra logic to exclude it since:
1. There was enforcement for retain to be at least 1.
1. The volume head will always have a newer timestamp.

We are accepting a 0 retain value for the snapshot-cleanup/delete, so I will exclude the volume head.

Reply:
Test result: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2991/